test/tests/problems/eigen_problem/ne.i

```diff
@@ -0,0 +1,71 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  xmin = 0
+  xmax = 10
+  ymin = 0
+  ymax = 10
+  elem_type = QUAD4
+  nx = 8
+  ny = 8
+[]
+
+# the minimum eigenvalue of this problem is 2*(PI/a)^2;
+# Its inverse is 0.5*(a/PI)^2 = 5.0660591821169. Here a is equal to 10.
+
+[Variables]
+  [./u]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
+
+[Kernels]
+  [./diff]
+    type = Diffusion
+    variable = u
+  [../]
+
+  [./rhs]
+    type = Reaction
+    variable = u
+    eigen_kernel = true
+  [../]
+[]
+
+[BCs]
+  [./homogeneous]
+    type = DirichletBC
+    variable = u
+    boundary = '0 1 2 3'
+    value = 0
+  [../]
+[]
+
+[Problem]
+  type = EigenProblem
+  eigen_problem_type = gen_non_hermitian
+  which_eigen_pairs = TARGET_MAGNITUDE
```

Comment:
@fdkong sorry for being picky. This will go into rattlesnake almost everywhere, so I want to make sure the input makes sense in a user's point of view. It would make more sense here to use `SMALLEST_MAGNITUDE` and make it the default value for the power method and PJFNK eigen executioner. Another separate comment is that: can we make `gen_non_hermitian` the default `eigen_problem_type`?